apps/main/src/dao/entities/user-gauge-weight-votes.ts

```
@@ -0,0 +1,52 @@
+import { curvejsValidationSuite } from '@/dao/entities/validation/curvejs-validation'
+import type { ChainId } from '@/dao/types/dao.types'
+import type { ChainParams, ChainQuery } from '@ui-kit/lib/model/query'
+import { queryFactory } from '@ui-kit/lib/model/query'
+import { useApiStore } from '@ui-kit/shared/useApiStore'
+
+const _fetchUserGaugeWeightVotes = async ({ chainId, userAddress }: ChainQuery<ChainId> & { userAddress: string }) => {
+  const curve = useApiStore.getState().curve
+
+  try {
+    const gaugeVoteWeightsRes = await curve!.dao.userGaugeVotes(userAddress)
+
+    const data = {
+      powerUsed: Number(gaugeVoteWeightsRes.powerUsed),
+      veCrvUsed: Number(gaugeVoteWeightsRes.veCrvUsed),
+      gauges: gaugeVoteWeightsRes.gauges
+        .map((gauge) => ({
+          userPower: Number(gauge.userPower),
+          userVeCrv: Number(gauge.userVeCrv),
+          userFutureVeCrv: Number(gauge.userFutureVeCrv),
+          expired: gauge.expired,
+          ...gauge.gaugeData,
+          rootGaugeAddress: '',
+          relativeWeight: Number(gauge.gaugeData.relativeWeight),
+          totalVeCrv: Number(gauge.gaugeData.totalVeCrv),
+          nextVoteTime: null,
+          canVote: true,
+        }))
+        .sort((a, b) => b.userPower - a.userPower),
+    }
+
+    return data
+  } catch (error) {
+    console.error(error)
```

let's let the query fail